src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -0,0 +1,197 @@
+            }
+
+            if (argsNative.delegateType == IntPtr.Zero)
+            {
+                throw new ArgumentNullException(nameof(argsNative.delegateType));
+            }
+
+            if (argsNative.functionHandle == IntPtr.Zero)
+            {
+                throw new ArgumentNullException(nameof(argsNative.functionHandle));
+            }
+
+            StringFormat stringFormat = (StringFormat) argsNative.stringFormat;
+
+            CreateDelegateArgs args = new CreateDelegateArgs()
+            {
+                assemblyPath = MarshalToString(stringFormat, argsNative.assemblyPath),
+                typeName     = MarshalToString(stringFormat, argsNative.typeName),
+                methodName   = MarshalToString(stringFormat, argsNative.methodName),
+                delegateType = MarshalToString(stringFormat, argsNative.delegateType),
+            };
+
+            return args;
+        }
+
+        /// <summary>
+        /// Native hosting entry point for creating a native delegate
+        /// </summary>
+        /// <param name="argsNative">Reference to a <see cref="CreateNativeDelegateArgs"/> instance</param>
+        [CLSCompliant(false)]
+        public unsafe static int CreateNativeDelegate(CreateNativeDelegateArgs argsNative)
+        {
+            CreateDelegateArgs args;
+
+            try
+            {
+                args = CheckAndMarshalNativeInput(argsNative);
+
+                Delegate d = CreateDelegate(ref args);
+
+                IntPtr functionPtr = Marshal.GetFunctionPointerForDelegate(d);
+
+                lock(s_Delegates)
+                {
+                    s_Delegates[functionPtr] = d;
+                }
+
+                Marshal.WriteIntPtr(argsNative.functionHandle, functionPtr);
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Native hosting entry point for releasing a created native delegate
+        /// </summary>
+        /// <param name="functionPtr">Native delegate to be released</param>
+        [CLSCompliant(false)]
+        public unsafe static int ReleaseNativeDelegate(IntPtr functionPtr)
+        {
+            try
+            {
+                lock(s_Delegates)
+                {
+                    s_Delegates.Remove(functionPtr);
+                }
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return 0;
+        }
+
+        private static Delegate CreateDelegate(ref CreateDelegateArgs args)
+        {
+            if (PathInternal.IsPartiallyQualified(args.assemblyPath))
+            {
+                throw new ArgumentException(SR.Argument_AbsolutePathRequired, nameof(args.assemblyPath));
+            }
+
+            // Throws
+            Assembly assembly = LoadAssembly(args.assemblyPath);
```

Comment:
I think we should call `Path.GetFullPath` to normalize the path... to make the lookup more reliable.

Reply:
Hmmm... I think it was there origianlly for normalization, but must have been removed when I added the throw for partial paths.